feature-pitches/010-compute-sessions/compute-sessions.md

```diff
@@ -0,0 +1,220 @@
+
+##### Priority 4: [for the future/out of scope] Search & select from preexisting RenkuLab environments
+
+#### User Flow: Start Session
+
+![breadboard-start-session.jpg](breadboard-start-session.jpg)
+
+#### User Flow: Configure Session [nice to have]
+
+![breadboard-configure-session.jpg](breadboard-configure-session.jpg)
+
+#### User Flow: Publish Session [for the future/out of scope]
+
+![breadboard-publish-session.jpg](breadboard-publish-session.jpg)
+
+#### User Flow: Update a Repo-Based Session [nice to have]
+
+![breadboard-update-repo-based-session.jpg](breadboard-update-repo-based-session.jpg)
+
+### Things not communicated via the User Flows
+
+#### Starting Assumptions
+
+- In this build, we will not change the project GitLab CI-based image build process (for 'Configure a Session from a Code repo')
+
+#### Key Points
+
+- When a user creates a new project, the project has no sessions until a user adds one.
+  - [nice to have] Provide shortcuts for adding a session from the RenkuLab default environments
+- A user can create as many sessions as they want in a project.
+- A user can run 1 session per environment per project (similar to our restriction now of 1 session per user per project)
+
+#### 1-click to start the Default Session
+
+- One session in a project should be the “default” session, so that we can still offer a 1-click ‘Start Session’ button for the project as a whole, which starts the default session.
+
+#### Permissions
+
+- **Creating sessions:** users with write access to the project should be allowed to create and configure sessions.
+- **Launching a session:** For the purposes of this pitch, anyone with access to the project can launch any session that has been added in the project.
+- **Viewing running sessions:** Only the user who launched a session can see its status.
+
+#### Inside the session
+
+The session is launched into the `/work` directory. The repositories and external storage sources are provisioned as sub-directories:
+
+```bash
+/work
+  /repository-1
+  /repository-2
+  /external-storage-1
+  /external-storage-2
+```
+
+### Revisiting the Motivating Problems
+
+- When I am starting a new project, I want to **start out with a basic compute environment** (for example with a pre-built data science stack), and I’ll decide what extra packages I need later.
+  - → Solved by offering RenkuLab built-in session “templates” (name TBD) as the fastest route to adding a session to a project
+- When I create a new Renku project, I want to get started right away with a pre-built environment- **I don’t want to have to wait for the environment image to be built** to get started.
+  - → Same as above
+- When I start a session, I want to **clearly understand what environment image the session will use and what is installed in it**, so I am not confused where the compute environment comes from.
```

Comment:
imho this doesn't really address the "clearly understand what is installed" part?